cmd/moq-example/server.cpp

```diff
@@ -0,0 +1,523 @@
+                          const qtransport::TransportRemote& remote) override {}
+
+    void cb_unannounce(qtransport::TransportConnId conn_id,
+                       uint64_t track_namespace_hash,
+                       std::optional<uint64_t> track_name_hash) {
+
+        if (track_name_hash.has_value()) { // subscribe done received
+            _logger->info << "Received subscribe done from conn_id: " << conn_id
+                           << "  for namespace hash: " << track_namespace_hash
+                           << " name hash: " << *track_name_hash
+                           << ", informational update only"
+                           << std::flush;
+        } else {
+            _logger->debug << "Received unannounce from conn_id: " << conn_id
+                           << "  for namespace hash: " << track_namespace_hash
+                           << ", removing all tracks associated with namespace"
+                           << std::flush;
+
+            for (auto track_alias: qserver_vars::announce_active[track_namespace_hash][conn_id]) {
+                auto ptd = qserver_vars::pub_subscribes[track_alias][conn_id];
+                if (ptd != nullptr) {
+                    _logger->info << "Received unannounce from conn_id: " << conn_id
+                                  << "  for namespace hash: " << track_namespace_hash
+                                  << ", removing track alias: " << track_alias
+                                  << std::flush;
+
+                    _moq_instance.lock()->unsubscribeTrack(conn_id, ptd);
+                }
+                qserver_vars::pub_subscribes[track_alias].erase(conn_id);
+                if (qserver_vars::pub_subscribes[track_alias].empty()) {
+                    qserver_vars::pub_subscribes.erase(track_alias);
+                }
+            }
+
+            qserver_vars::announce_active[track_namespace_hash].erase(conn_id);
+            if (qserver_vars::announce_active[track_namespace_hash].empty()) {
+                qserver_vars::announce_active.erase(track_namespace_hash);
+            }
+        }
+    }
+
+    bool cb_announce(qtransport::TransportConnId conn_id,
+                     uint64_t track_namespace_hash) override {
+
+        _logger->debug << "Received announce from conn_id: " << conn_id
+                       << "  for namespace_hash: " << track_namespace_hash
+                       << std::flush;
+
+        // Add to state if not exist
+        auto [anno_conn_it, is_new] = qserver_vars::announce_active[track_namespace_hash].try_emplace(conn_id);
+
+        if (!is_new) {
+            _logger->info << "Received announce from conn_id: " << conn_id
+                          << " for namespace_hash: " << track_namespace_hash
+                          << " is duplicate, ignoring" << std::flush;
+            return true;
+        }
+
+        // true results in Send announce OK
+        return true;
+    }
```

Comment:
May not matter, but this never returns false?

Reply:
Yeah, it doesn't matter right now.  It's for the future if we did some auth/validation, which would then return false.  Same thing for subscribe.